R/clean_ProteinProspector.R

```diff
@@ -0,0 +1,45 @@
+#' Clean raw Protein Prospector data
+#' @param msstats_object object that inherits from MSstatsInputFiles class.
+#' @return data.table
+#' @keywords internal
+#' @noRd
+.cleanRawProteinProspector = function(msstats_object) {
+    protein_prospector_input = getInputFile(msstats_object, "input")
+    protein_prospector_input = 
+        data.table::as.data.table(protein_prospector_input)
+    channels = .getChannelColumns(
+        colnames(protein_prospector_input), "Int")
+    req_cols = c('AccX', 'z', 'DBPeptide', 'Fraction', channels)
+    protein_prospector_input = protein_prospector_input[, req_cols, with = FALSE]
+    data.table::setnames(
+        protein_prospector_input, 
+        c("AccX", "DBPeptide", "z", "Fraction"),
+        c("ProteinName", "PeptideSequence", "PrecursorCharge", "Run"), 
+        skip_absent = TRUE)
+    protein_prospector_input$PSM = paste(
+        protein_prospector_input$PeptideSequence, 
+        protein_prospector_input$PrecursorCharge,
+        1:nrow(protein_prospector_input), sep = "_"
+    )
+    
+    protein_prospector_input = melt(protein_prospector_input, 
+                                    measure.vars = channels, 
+                                    id.vars = setdiff(
+                                        colnames(protein_prospector_input), 
+                                        channels
+                                    ),
+                                    variable.name = "Channel", 
+                                    value.name = "Intensity"
```

Comment:
would it be useful to variable.factor = FALSE?

Reply:
I'm not sure if there'll be any added benefit.  

I also tested on the dataset provided to see if there's any difference in space usage, and using factors reduced the memory usage of the channels column by 50% (8112 bytes -> 4680 bytes of space stored for the Channel column).  This seems to be supported by this [post](https://stackoverflow.com/questions/74608288/in-r-do-factors-somehow-save-space) since there's many duplicates.

Reply:
right, the integer representation helps, let's leave it as it is